mongodbatlas/performance_advisor.go

```diff
@@ -0,0 +1,230 @@
+type Operation struct {
+	Raw        string                         `json:"raw,omitempty"`        // Raw log line produced by the query.
+	Stats      Stats                          `json:"stats,omitempty"`      // Query statistics.
+	Predicates []map[string]map[string]string `json:"predicates,omitempty"` // Documents containing the search criteria used by the query.
+}
+
+// Stats represents query statistics.
+type Stats struct {
+	MS        float64 `json:"ms,omitempty"`        // Duration in milliseconds of the query.
+	NReturned int64   `json:"nReturned,omitempty"` // Number of results returned by the query.
+	NScanned  int64   `json:"nScanned,omitempty"`  // Number of documents read by the query.
+	TS        int64   `json:"ts,omitempty"`        // Query timestamp, in seconds since epoch.
+}
+
+// SuggestedIndex represents a suggested index.
+type SuggestedIndex struct {
+	ID        string           `json:"id,omitempty"`        // Unique id for this suggested index.
+	Impact    []string         `json:"impact,omitempty"`    // List of unique identifiers which correspond the query shapes in this response which pertain to this suggested index.
+	Namespace string           `json:"namespace,omitempty"` // 	Namespace of the suggested index.
+	Weight    float64          `json:"weight,omitempty"`    // Estimated percentage performance improvement that the suggested index would provide.
+	Index     []map[string]int `json:"index,omitempty"`     // Array of documents that specifies a key in the index and its sort order, ascending or descending.
+}
+
+// SuggestedIndexes represents an array of suggested indexes.
+type SuggestedIndexes struct {
+	SuggestedIndexes []*SuggestedIndex `json:"suggestedIndexes,omitempty"` // Documents with information about the indexes suggested by the Performance Advisor.
+	Shapes           []*Shape          `json:"shapes,omitempty"`           // Documents with information about the query shapes that are served by the suggested indexes.
+
+}
+
+// SlowQueriesOptions contains the request query parameters for the API request.
+type SlowQueriesOptions struct {
+	Since      int64  `url:"since,omitempty"`      // Point in time, specified as milliseconds since the Unix Epoch, from which you want to receive results.
+	Duration   int64  `url:"duration,omitempty"`   // 	Length of time from the since parameter, in milliseconds, for which you want to receive results.
+	Envelope   bool   `url:"envelope,omitempty"`   // Boolean that specifies whether or not to wrap the response in an envelope. Default is false.
+	Namespaces string `url:"namespaces,omitempty"` // Namespaces from which to retrieve slow query logs. A namespace consists of the database and collection resource separated by a ., such as <database>.<collection>.
+	NLogs      int64  `url:"nLogs,omitempty"`      // Maximum number of log lines to return. Defaults to 20000.
+	Pretty     bool   `url:"pretty,omitempty"`     // Display response in a prettyprint format. Defaults to false.
```

Comment:
similar to envelope this is not supported, mainly as it has no meaning in for the client, nothing would change